File: app/src/main/java/com/kevintu/annotation/base/BaseActivity.java
package com.kevintu.annotation.base;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Create by Kevin-Tu on 2019/10/11.
 */
public class BaseActivity extends AppCompatActivity {

    private Unbinder unbinder;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (openButterKnife()) {
            unbinder = ButterKnife.bind(this);
        }
    }

    public boolean openButterKnife() {
        return false;
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (unbinder != null) {
            unbinder.unbind();
        }
    }
}

File: app/src/main/java/com/kevintu/annotation/base/BaseActivity.kt
package com.kevintu.annotation.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import butterknife.ButterKnife
import butterknife.Unbinder

/**
 * Create by Kevin-Tu on 2019/10/10.
 */
open class BaseActivity : AppCompatActivity() {

    private var unbinder: Unbinder? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (openButterKnife()) {
            unbinder = ButterKnife.bind(this)
        }
    }

    open fun openButterKnife() : Boolean {
        return false
    }

    override fun onDestroy() {
        super.onDestroy()
        unbinder?.unbind()
    }
}